### PR TITLE
Support new schema of deployment.json with backward compatibility

### PR DESCRIPTION
**Pull request by LazarusX**
This PR implements https://github.com/Azure/iotedgedev/issues/212 and fixes https://github.com/Azure/iotedgedev/issues/213.

### CODE REVIEW

File: iotedgedev/deploymentmanifest.py [hunk 1]
```diff
@@ -52,24 +56,58 @@ def add_default_route(self, module_name):
         new_route_name = "{0}ToIoTHub".format(module_name)
         new_route = "FROM /messages/modules/{0}/outputs/* INTO $upstream".format(module_name)
 
-        self.utility.nested_set(self.json, ["moduleContent", "$edgeHub", "properties.desired", "routes", new_route_name], new_route)
+        try:
+            self.utility.nested_set(self.get_module_content(), ["$edgeHub", "properties.desired", "routes", new_route_name], new_route)
```
Comment (jongio) [inline iotedgedev/deploymentmanifest.py]: Seems like we can further abstract "get desired property" which takes module and name.

File: tests/assets/deployment.template_2.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
           }
         },
         "modules": {
-          "temp-sensor-module": {
+          "tempSensor": {
```
Comment (jongio) [inline tests/assets/deployment.template_2.json]: I used all lower because docker is all lower and I use that as the docker name.  Thoughts?

Comment (LazarusX) [reply tests/assets/deployment.template_2.json]: @jongio The documents all reference to the temp sensor module as "tempSensor", for example https://docs.microsoft.com/en-us/azure/iot-edge/quickstart-linux#deploy-a-module and https://docs.microsoft.com/en-us/azure/iot-edge/tutorial-csharp-module#build-your-iot-edge-solution. For container names, Docker supports upper case. The edgeAgent and edgeHub also follows this style.

Comment (jongio) [reply tests/assets/deployment.template_2.json]: @LazarusX  - This is what I get when I run build with mixed case

```
(venv) C:\temp>docker build . -t testFile
invalid argument "testFile" for "-t, --tag" flag: invalid reference format: repository name must be lowercase
See 'docker build --help'.
```

File: iotedgedev/deploymentmanifest.py (hunk 1 above)
Comment (LazarusX) [reply iotedgedev/deploymentmanifest.py]: @jongio Added. Thanks.

File: tests/assets/deployment.template_2.json (hunk 2 above)
Comment (jongio) [reply tests/assets/deployment.template_2.json]: Okay.

**Comment by jongio** [comment]
It would be good to add tests for the new get_desired_property method.

**Comment by LazarusX** [comment]
@jongio Tests added. Thanks.

**Comment by jongio** [comment]
@LazarusX  - you ready for merge on this?

**Comment by LazarusX** [comment]
@jongio Yes, you can merge it if it looks good to you.

**Pull request merged**